Name : Rattanavilay, Thip<br>
Class : DSC530<br>
Week 8: Hypothesis Testing & Linear Least Squares<br>
Assignment : 8.2, 9-1<br>

**As sample size increases, the power of a hypothesis test increases, which means it is more likely to be positive if the effect is real. Conversely, as sample size decreases, the test is less likely to be positive even if the effect is real.**

**To investigate this behavior, run the tests in this chapter with different subsets of the NSFG data. 
You can use thinkstats2.SampleRows to select a random subset of the rows in a DataFrame.**

**What happens to the p-values of these tests as sample size decreases? 
What is the smallest sample size that yields a positive test?**



In [1]:
import first
import hypothesis
import scatter
import thinkstats2

import numpy as np

In [2]:
def RunTests(live, iters=1000):
    """Runs the tests from Chapter 9 with a subset of the data.

    live: DataFrame
    iters: how many iterations to run
    """
    n = len(live)
    firsts = live[live.birthord == 1]
    others = live[live.birthord != 1]

    # compare pregnancy lengths
    data = firsts.prglngth.values, others.prglngth.values
    ht = hypothesis.DiffMeansPermute(data) # difference in means by permutation
    p1 = ht.PValue(iters=iters)            # Get the p-value

    # test correlation
    live2 = live.dropna(subset=['agepreg', 'totalwgt_lb'])
    data = live2.agepreg.values, live2.totalwgt_lb.values
    ht = hypothesis.CorrelationPermute(data)  # correlations by permutation
    p3 = ht.PValue(iters=iters)               # Get the p-value
    

    print('%d\t%0.2f\t%0.2f' % (n, p1, p3))

In [5]:
thinkstats2.RandomSeed(18)

live, firsts, others = first.MakeFrames()

print('n\tP_Len\tCorr')
n = len(live)
for _ in range(7):      # iterate 7 times
    sample = thinkstats2.SampleRows(live, n)
    RunTests(sample)
    n //= 2

n	P_Len	Corr
9148	0.16	0.00
4574	0.63	0.00
2287	0.55	0.00
1143	0.25	0.00
571	0.29	0.10
285	0.55	0.01
142	0.36	0.63


Conclusion:<br>

As you can see here I have ran 2 tests and P-values are shown above for corresponding sample size,n :<br>
P_Len - Difference in mean pregnancy length<br>
Corr - Correlation between mother's age and birth weight<br>
 
Shown above here are the outccome of 2 tests we can clearly see that, power of hypothesis increases with size of sample. Test is positive with large sample size.<br>
 
In this test, smallest size is '285' which yields positive test